### ISSUE TITLE
Taildrop "connection refused" when trying to send file to Android phone

### ISSUE THREAD

**Issue by biggerfisch** [tailscale/tailscale#3244]
### What is the issue?

I'm trying to send a file to my new Android phone, running Android 12, from several different linux systems, but I always get this error: 
```
$ sudo tailscale file cp ./me.png <device_name>:
Put "http://<IP>:38601/v0/put/me.png": dial tcp 100.81.97.3:38601: connect: connection refused
```

The phone is connected to tailscale and can send files to an iOS device, but the iOS device cannot send files to the Android phone, getting an error of "Failed to share file with <device_name> Please reconnect <devicename> to Tailscale and try again" (the Android phone is actually connected and on the same wifi network as the iOS device).

### Steps to reproduce

1. Have a new Google Pixel 6 Pro (runs Android 12 by default)
2. Connect it to Tailscale
3. Try to send a file to the device from either iOS or Linux
4. Get error message

### Are there any recent changes that introduced the issue?

Android 12 is rather new, possible this.

### OS

Linux, iOS, Android

### OS version

Android 12

### Tailscale version

1.16.1 (Android), 1.16.1 (iOS, Linux), 1.16.2 (Linux)

### Bug report

BUG-a0d7655a7c29a4a8926b2e467c9671e2217b87d4c9a86d6e9f62894678320c0c-20211102170619Z-32ad4e4bb86d2b95

**Comment by DentonGentry** [comment]
I don't think it is Android 12 specifically, I'm running Android 12 on a Pixel 3a and able to transfer files to it.

----

LogIds
- of the Android device in this bug: a28780a8fc92e3c47c2452b58844035f93b3a8109af8a6725e6f85d7e6055649
- iOS device: 34fee3e30a374dff1d82b65bc4b6ad069adb9ad839d425ea18d63d745b0c0912

**Comment by kermieisinthehouse** [comment]
I run into this on Android 11

**Comment by JulesMike** [comment]
Running Android 11 here and having same issue as OP.

Tried sending a file from linux (5.13.19-2-MANJARO):
```shell
$ sudo tailscale file cp ./image.gif my-phone:
Put "http://<IP>:52140/v0/put/image.gif": dial tcp <IP>:52140: connect: connection refused
```

**Bug Report**
BUG-594c30833fb4abfe64b9f19f5f4d1728b5450bf1d97ae3a1f6bbc4f08b25f841-20211214002548Z-cf3dd745612df683

**Comment by DentonGentry** [comment]
Looking at `BUG-594c30833fb4abfe64b9f19f5f4d1728b5450bf1d97ae3a1f6bbc4f08b25f841-20211214002548Z-cf3dd745612df683`:

On the Manjaro Linux machine initiating the transfer:
```
2021-12-14 04:23:33.147522258 +0400 +0400: [5UmO2] - Sending handshake initiation
2021-12-14 04:23:33.179693141 +0400 +0400: peer keys: [5UmO2]
2021-12-14 04:23:33.179749872 +0400 +0400: v1.18.1 peers: 27763964/694600
2021-12-14 04:23:33.179947182 +0400 +0400: [5UmO2] - Received handshake response
2021-12-14 04:23:33.190626449 +0400 +0400: open-conn-track: flow TCP (TCP 100.111.a.b:51332 => 100.76.c.d:52140) got RST by peer
```

On the Android 11 device on the same account, at about that same time:
```
2021-12-14 00:24:23.741230936 +0000 UTC: 7.7M/241.9M [j8RoV] - Received handshake initiation
2021-12-14 00:24:23.741835705 +0000 UTC: 7.7M/241.9M [j8RoV] - Sending handshake response
2021-12-14 00:24:23.742374089 +0000 UTC: 7.7M/241.9M v1.18.0-t71f1dd5aa-g6e464cc68f9 peers: 629752/28009308
2021-12-14 00:24:23.752355243 +0000 UTC: 7.7M/241.9M Accept: TCP{100.111.a.b:51332 > 100.76.c.d:52140} 60 tcp ok
2021-12-14 00:24:23.754439743 +0000 UTC: 7.8M/241.9M Accept: TCP{100.76.c.d:52140 > 100.111.a.b:51332} 40 ok out
```

The two systems have timezones set differently, and it looks like their clocks are about 50 seconds off. However, the Android device thinks it send an OK while the Manjaro device received an RST.

**Comment by JulesMike** [comment]
@DentonGentry thanks for prompt response.

I do confirm after manually setting my timezone on my Android device to match the linux machine, file transfer worked flawlessly :)

Do note it worked before when sending from the Android device to the linux machine.

**Comment by DentonGentry** [comment]
Wait, what? I didn't expect setting the timezone to have an effect, just noted it because the timestamps were different in the telemetry I was pasting.

Ok, that is weird.

**Comment by JulesMike** [comment]
More information:
I just found out I have the same issue on my Android 10 device (Samsung s9 running custom ROM) when sending file to it from the Manjaro linux machine. The time zone also there (on the Android device) was set to "Automatic - Set by network". I manually changed the timezone to "GMT+0400 - Mauritius" (as on my linux machine), disconnected/reconnected in tailscale and it worked.

**Comment by JulesMike** [comment]
New update:
I've set the timezone back to "Automatic - Set by network" on my Android 10 device, tried again and file transfer worked now.
Now this is weird 😅 

**Bug report**:
BUG-594c30833fb4abfe64b9f19f5f4d1728b5450bf1d97ae3a1f6bbc4f08b25f841-20211214012025Z-f4813845a480b852

**Comment by biggerfisch** [comment]
Can confirm, disabling "set timezone automatically" fixed it so far, even after reenabling the setting

**Comment by JulesMike** [comment]
@biggerfisch I re enabled "set timezone automatically" on my Android 10 and 11 devices and it still works on my side. Can you confirm on your Android 12 device?

**Comment by biggerfisch** [comment]
Yup, works, even after restarting my Android 12 device. That said... my app _did_ update to 1.18 since the last time I tried this and it cannot be ruled out that that accidentally fixed this issue instead.

What I can say "for sure" is that the current process of:
1. Have 1.18 app (1.16.2 debian box)
2. Disable "set timezone automatically"
3. Send file
4. Enable "set timezone automatically"

seems to end up in a working state.

**Comment by JulesMike** [comment]
I encountered the issue today on Tailscale 1.18.

**Comment by DentonGentry** [comment]
I believe this was fixed in the 1.22 release, by making the peerapi use netstack.
